python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -826,6 +856,26 @@ def do_evaluate(
                     else prefix.obj,
                 )
             )
+        elif self.name == pl_expr.StringFunction.Replace:
+            column, target, repl = columns
+            n, _ = self.options
+            return Column(
+                plc.strings.replace.replace(
+                    column.obj, target.obj_scalar, repl.obj_scalar, maxrepl=n
+                )
+            )
+        elif self.name == pl_expr.StringFunction.ReplaceMany:
+            column, target, repl = columns
+            # if plc.binaryop.binary_operation(target.obj):
+            #     # libcudf ignores empty strings for replace_many
+            #     # polars inserts the replacement after every character
+            #     raise NotImplementedError(
+            #         "libcudf replace_many is implemented differently from polars "
+            #         "for empty strings"
+            #     )
```

Comment:
What happened here? (I set to DO NOT MERGE until resolved)

Reply:
Sorry, I made a mess here.

If you have time (later of course), can you take another look at this PR?

(the main changes made were rejecting non literalcolumn inputs to replacemany and rejecting empty strings in the targetcolumn for replacemany).